包名为中文的时候toJSONString会有异常

试了最新的几个版本都不行啊，包名为中文字符的时候就异常了，UnsupportedOperationException

`2020-04-22 16:59:02.062 ERROR 56072 --- [nio-8080-exec-2] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is com.alibaba.fastjson.JSONException: create asm serializer error, verson 1.2.58, class class 中文包名.Money] with root cause

java.lang.UnsupportedOperationException: null
	at com.alibaba.fastjson.asm.ByteVector.putUTF8(ByteVector.java:191) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.asm.ClassWriter.newUTF8(ClassWriter.java:272) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.asm.ClassWriter.newClassItem(ClassWriter.java:283) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.asm.ClassWriter.newConstItem(ClassWriter.java:262) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.asm.MethodWriter.visitLdcInsn(MethodWriter.java:217) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.serializer.ASMSerializerFactory.createJavaBeanSerializer(ASMSerializerFactory.java:199) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.serializer.SerializeConfig.createASMSerializer(SerializeConfig.java:90) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.serializer.SerializeConfig.createJavaBeanSerializer(SerializeConfig.java:242) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.serializer.SerializeConfig.createJavaBeanSerializer(SerializeConfig.java:118) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.serializer.SerializeConfig.getObjectWriter(SerializeConfig.java:776) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.serializer.SerializeConfig.getObjectWriter(SerializeConfig.java:430) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.serializer.JSONSerializer.getObjectWriter(JSONSerializer.java:410) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.serializer.JSONSerializer.write(JSONSerializer.java:282) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.JSON.toJSONString(JSON.java:740) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.JSON.toJSONString(JSON.java:678) ~[fastjson-1.2.58.jar:na]
	at com.alibaba.fastjson.JSON.toJSONString(JSON.java:643) ~[fastjson-1.2.58.jar:na]
	at 中文包名.MoneyController.test(MoneyController.java:24) ~[classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_181]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_181]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_181]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_181]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190) ~[spring-web-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138) ~[spring-web-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:879) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:793) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:898) ~[spring-webmvc-5.2.5.RELEASE.jar:5.2.5.RELEASE]
`

> 如果可以的话,请贴出你的代码!

新建一个maven项目，引入fastjson
        <dependency>
            <groupId>com.alibaba</groupId>
            <artifactId>fastjson</artifactId>
            <version>1.2.68</version>
        </dependency>

一个简单的代码：
package 我是个中文包名;
import com.alibaba.fastjson.JSONObject;
/**
 * @author zhl
 * @date 2020/4/23 0023 10:38
 */
public class User {
    Integer id ;
    String name;
    public Integer getId() {
        return id;
    }
    public void setId(Integer id) {
        this.id = id;
    }
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
    public static void main(String[] args) {
        User user = new User();
        user.setId(9);
        user.setName("asdffsf");
        System.out.println(JSONObject.toJSONString(user));
    }
}

输出：
Exception in thread "main" com.alibaba.fastjson.JSONException: create asm serializer error, verson 1.2.68, class class 我是个中文包名.User
	at com.alibaba.fastjson.serializer.SerializeConfig.createJavaBeanSerializer(SerializeConfig.java:264)
	at com.alibaba.fastjson.serializer.SerializeConfig.createJavaBeanSerializer(SerializeConfig.java:120)
	at com.alibaba.fastjson.serializer.SerializeConfig.getObjectWriter(SerializeConfig.java:792)
	at com.alibaba.fastjson.serializer.SerializeConfig.getObjectWriter(SerializeConfig.java:436)
	at com.alibaba.fastjson.serializer.JSONSerializer.getObjectWriter(JSONSerializer.java:416)
	at com.alibaba.fastjson.serializer.JSONSerializer.write(JSONSerializer.java:282)
	at com.alibaba.fastjson.JSON.toJSONString(JSON.java:758)
	at com.alibaba.fastjson.JSON.toJSONString(JSON.java:696)
	at com.alibaba.fastjson.JSON.toJSONString(JSON.java:661)
	at 我是个中文包名.User.main(User.java:28)
Caused by: java.lang.UnsupportedOperationException
	at com.alibaba.fastjson.asm.ByteVector.putUTF8(ByteVector.java:191)
	at com.alibaba.fastjson.asm.ClassWriter.newUTF8(ClassWriter.java:272)
	at com.alibaba.fastjson.asm.ClassWriter.newClassItem(ClassWriter.java:283)
	at com.alibaba.fastjson.asm.ClassWriter.newConstItem(ClassWriter.java:262)
	at com.alibaba.fastjson.asm.MethodWriter.visitLdcInsn(MethodWriter.java:218)
	at com.alibaba.fastjson.serializer.ASMSerializerFactory.createJavaBeanSerializer(ASMSerializerFactory.java:199)
	at com.alibaba.fastjson.serializer.SerializeConfig.createASMSerializer(SerializeConfig.java:92)
	at com.alibaba.fastjson.serializer.SerializeConfig.createJavaBeanSerializer(SerializeConfig.java:248)
	... 9 more

Process finished with exit code 1

--- a/src/main/java/com/alibaba/fastjson/asm/ByteVector.java
+++ b/src/main/java/com/alibaba/fastjson/asm/ByteVector.java
@@ -185,7 +185,7 @@ public class ByteVector {
 		data[len++] = (byte) charLength;
 		for (int i = 0; i < charLength; ++i) {
 			final char c = s.charAt(i);
-			if (c >= '\001' && c <= '\177') {
+			if ((c >= '\001' && c <= '\177') || (c >= '\u4E00' && c <= '\u9FFF')) {
 				data[len++] = (byte) c;
 			} else {
 				throw new UnsupportedOperationException();

--- a/src/main/java/com/alibaba/fastjson/parser/deserializer/JavaBeanDeserializer.java
+++ b/src/main/java/com/alibaba/fastjson/parser/deserializer/JavaBeanDeserializer.java
@@ -510,18 +510,18 @@ public class JavaBeanDeserializer implements ObjectDeserializer {
             String typeKey = beanInfo.typeKey;
             for (int fieldIndex = 0, notMatchCount = 0;; fieldIndex++) {
                 String key = null;
-                FieldDeserializer fieldDeser = null;
+                FieldDeserializer fieldDeserializer = null;
                 FieldInfo fieldInfo = null;
                 Class<?> fieldClass = null;
-                JSONField feildAnnotation = null;
-                boolean customDeserilizer = false;
+                JSONField fieldAnnotation = null;
+                boolean customDeserializer = false;
                 if (fieldIndex < sortedFieldDeserializers.length && notMatchCount < 16) {
-                    fieldDeser = sortedFieldDeserializers[fieldIndex];
-                    fieldInfo = fieldDeser.fieldInfo;
+                    fieldDeserializer = sortedFieldDeserializers[fieldIndex];
+                    fieldInfo = fieldDeserializer.fieldInfo;
                     fieldClass = fieldInfo.fieldClass;
-                    feildAnnotation = fieldInfo.getAnnotation();
-                    if (feildAnnotation != null && fieldDeser instanceof DefaultFieldDeserializer) {
-                        customDeserilizer = ((DefaultFieldDeserializer) fieldDeser).customDeserilizer;
+                    fieldAnnotation = fieldInfo.getAnnotation();
+                    if (fieldAnnotation != null && fieldDeserializer instanceof DefaultFieldDeserializer) {
+                        customDeserializer = ((DefaultFieldDeserializer) fieldDeserializer).customDeserilizer;
                     }
                 }
 
@@ -529,9 +529,9 @@ public class JavaBeanDeserializer implements ObjectDeserializer {
                 boolean valueParsed = false;
                 
                 Object fieldValue = null;
-                if (fieldDeser != null) {
+                if (fieldDeserializer != null) {
                     char[] name_chars = fieldInfo.name_chars;
-                    if (customDeserilizer && lexer.matchField(name_chars)) {
+                    if (customDeserializer && lexer.matchField(name_chars)) {
                         matchField = true;
                     } else if (fieldClass == int.class || fieldClass == Integer.class) {
                         int intVal = lexer.scanFieldInt(name_chars);
@@ -651,10 +651,10 @@ public class JavaBeanDeserializer implements ObjectDeserializer {
                         }
                     } else if (fieldClass.isEnum() // 
                             && parser.getConfig().getDeserializer(fieldClass) instanceof EnumDeserializer
-                            && (feildAnnotation == null || feildAnnotation.deserializeUsing() == Void.class)
+                            && (fieldAnnotation == null || fieldAnnotation.deserializeUsing() == Void.class)
                             ) {
-                        if (fieldDeser instanceof DefaultFieldDeserializer) {
-                            ObjectDeserializer fieldValueDeserilizer = ((DefaultFieldDeserializer) fieldDeser).fieldValueDeserilizer;
+                        if (fieldDeserializer instanceof DefaultFieldDeserializer) {
+                            ObjectDeserializer fieldValueDeserilizer = ((DefaultFieldDeserializer) fieldDeserializer).fieldValueDeserilizer;
                             fieldValue = this.scanEnum(lexer, name_chars, fieldValueDeserilizer);
 
                             if (lexer.matchStat > 0) {
@@ -835,7 +835,7 @@ public class JavaBeanDeserializer implements ObjectDeserializer {
 
                 if (matchField) {
                     if (!valueParsed) {
-                        fieldDeser.parseField(parser, object, type, fieldValues);
+                        fieldDeserializer.parseField(parser, object, type, fieldValues);
                     } else {
                         if (object == null) {
                             fieldValues.put(fieldInfo.name, fieldValue);
@@ -846,10 +846,10 @@ public class JavaBeanDeserializer implements ObjectDeserializer {
                                     && fieldClass != double.class //
                                     && fieldClass != boolean.class //
                                     ) {
-                                fieldDeser.setValue(object, fieldValue);
+                                fieldDeserializer.setValue(object, fieldValue);
                             }
                         } else {
-                            fieldDeser.setValue(object, fieldValue);
+                            fieldDeserializer.setValue(object, fieldValue);
                         }
 
                         if (setFlags != null) {

--- a/src/test/java/com/alibaba/json/bvt/issue_3000/Issue3132.java
+++ b/src/test/java/com/alibaba/json/bvt/issue_3000/Issue3132.java
@@ -1,0 +1,16 @@
+package com.alibaba.json.bvt.issue_3000;
+
+import com.alibaba.fastjson.JSONObject;
+import com.alibaba.json.bvtVO.一个中文名字的包.User;
+import junit.framework.TestCase;
+
+public class Issue3132 extends TestCase {
+
+    public void test_for_issue() throws Exception {
+        User user = new User();
+        user.setId(9);
+        user.setName("asdffsf");
+        System.out.println(JSONObject.toJSONString(user));
+    }
+
+}

--- a/src/test/java/com/alibaba/json/bvtVO/一个中文名字的包/User.java
+++ b/src/test/java/com/alibaba/json/bvtVO/一个中文名字的包/User.java
@@ -1,0 +1,18 @@
+package com.alibaba.json.bvtVO.一个中文名字的包;
+
+public class User {
+    Integer id ;
+    String name;
+    public Integer getId() {
+        return id;
+    }
+    public void setId(Integer id) {
+        this.id = id;
+    }
+    public String getName() {
+        return name;
+    }
+    public void setName(String name) {
+        this.name = name;
+    }
+}

